src/data/sonata_data.cpp

```diff
@@ -98,13 +98,16 @@ void SonataData::prepare_buffer(size_t max_buffer_size) {
 }
 
 bool SonataData::is_due_to_report(double step) const noexcept {
-    // Dont record data if current step < tstart
-    if (step < last_step_recorded_) {
+    // Don't record data if there are no steps (i.e., 'Duration 0')
+    if (last_step_ == 0) {
```

Comment:
@jorblancoa Perhaps there is a better way to handle this matter compare to hiding the issue with the sanity-check. Let's have a discussion when you get back, ok?